src/nodeProvider.ts

```diff
@@ -38,21 +40,23 @@ export class DidactNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 	}
 
 	// get the list of children for the node provider
-	public getChildren(element?: TreeNode): TreeNode[] {
-        if (!element) {
-            return this.treeNodes;
-        } else {
-            // assume it's a category
-            if (element.label && !(element instanceof TutorialNode)) {
+	public async getChildren(element?: any): Promise<any[]> {
```

Comment:
because we don't just return TreeNodes - they're HeadingNodes or TutorialNodes. TutorialNode is a TreeNode, but HeadingNode is actually a vscode.TreeItem. it's a mix of things

Reply:
Will try and clean things up with https://issues.redhat.com/browse/FUSETOOLS2-1050

Reply:
if there are two types, TypeScript is providign th epossibility to provide both of them with `HeadingNodes | TutorialNodes`

it sounds better to me to handle it inside this PR as it is introducing technical debt that sounds ot be easuily avoided. But fine to handle it in another task as longas it is done shortly after.

Reply:
@apupier it's more than that, so I will handle it in a separate follow-on PR